### PR TITLE
Add some convenient functions for protecting views in custom installations

### PR DESCRIPTION
**Pull request by timthelion**
Not sure if this needs to be merged into the main project but it sure would be convenient for me :) . Client needed to paywall the ticketing system and this was the easiest way ;)

### CODE REVIEW

**Review by Benbb96 — CHANGES_REQUESTED** [review]
I think it makes sense to add these possible checks in decorators. Could you just add the related documentation in order to show how they might be used, please?

The setting `VALID_EXTENSIONS` is documented but isn't present in `settings.py` so I believe it should be removed in profit for your `HELPDESK_VALIDATE_ATTACHMENT_TYPES`.

The tests also need to be adapted for this new setting I think.

**Comment by timthelion** [comment]
Oops, I didn't realize that `VALID_EXTENSIONS` was documented. I'll update the docs, but not this week ;) No time...

**Review by Benbb96 — DISMISSED** [review]
All good for me!

**Comment by timthelion** [comment]
@uhurusurfa Can you merge this please?

File: docs/settings.rst [hunk 1]
```diff
@@ -85,18 +103,10 @@ These changes are visible throughout django-helpdesk
 
   **Default:** ``HELPDESK_MAX_EMAIL_ATTACHMENT_SIZE = 512000``
 
-- **VALID_EXTENSIONS** Valid extensions for file types that can be attached to tickets
-
-  **Default:** ``VALID_EXTENSIONS = ['.txt', '.asc', '.htm', '.html', '.pdf', '.doc', '.docx', '.odt', '.jpg', '.png', '.eml']
-
```
Comment (uhurusurfa) [inline docs/settings.rst:91]: Mark this as deprecated but continue to support it until some futiure release to maintain backwards compatibility.

File: helpdesk/settings.py [hunk 2]
```diff
@@ -329,6 +337,10 @@
 # Override it in your own Django settings.py
 HELPDESK_ATTACHMENT_DIR_PERMS = int(getattr(settings, 'HELPDESK_ATTACHMENT_DIR_PERMS', "755"), 8)
 
+HELPDESK_VALID_EXTENSIONS = getattr(settings, 'HELPDESK_VALID_EXTENSIONS', ['.txt', '.asc', '.htm', '.html', '.pdf', '.doc', '.docx', '.odt', '.jpg', '.png', '.eml'])
+
+HELPDESK_VALIDATE_ATTACHMENT_TYPES = getattr(settings, 'HELPDESK_VALIDATE_ATTACHMENT_TYPES', True)
+
```
Comment (uhurusurfa) [inline helpdesk/settings.py:343]: Use `VALID_EXTENSIONS` if it exists and fall back to `HELPDESK_VALID_EXTENSIONS` if empty to maintain backwards compatibility.

Failing there will ned to be clear upgrade notes telling users to change the name when they upgrade.

**Review by uhurusurfa — CHANGES_REQUESTED** [review]
See inline comments

File: helpdesk/views/api.py [hunk 3]
```diff
@@ -34,6 +36,10 @@ def get_queryset(self):
 class TicketViewSet(viewsets.ModelViewSet):
     """
     A viewset that provides the standard actions to handle Ticket
+
+    You can filter the tickets by status using the `status` query parameter. For example:
+
+    `/api/tickets/?status=Open,Resolved` will return all the tickets that are Open or Resolved.
```
[the remark this answers is not in the document]
Comment (timthelion) [reply helpdesk/views/api.py:42]: Somehow the idea of passing arbitrary numbers just felt less nice than having the names but you're right about the localization issue so I'll change it.

**Review by uhurusurfa — APPROVED** [review]

**Pull request merged**